backend/drm/drm.c

```diff
@@ -1629,6 +1630,26 @@ void destroy_drm_connector(struct wlr_drm_connector *conn) {
 	free(conn);
 }
 
+int wlr_drm_backend_get_client_fd(struct wlr_backend *backend) {
+	assert(backend && wlr_backend_is_drm(backend));
+
+	struct wlr_drm_backend *drm = get_drm_backend_from_backend(backend);
+	char *path = drmGetDeviceNameFromFd2(drm->fd);
+
+	int fd = open(path, O_RDWR);
+	if (fd < 0) {
+		wlr_log_errno(WLR_ERROR, "Unable to clone DRM fd for client fd.\n");
+		free(path);
+		return -EINVAL;
```

Comment:
`EINVAL` isn't meaningful here.

If callers don't care about the error cause, we can just return -1 on error.